### PR TITLE
Issue-567 Adding button templates

### PR DESCRIPTION
**Pull request by althalus86**
### Reasons for making this change

Exposing the AddButton and IconButton to be able to replace them with a template assigned from the Form properties as AddButtonTemplate, RemoveButtonTempalte, MoveUpButtonTemplate and MoveDownButtonTemplate

Fixes https://github.com/mozilla-services/react-jsonschema-form/issues/567

If this is related to existing tickets, include links to them as well.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've checked the rendering of the Markdown text I've added
  - [ ] If I'm adding a new section, I've updated the Table of Content
* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated docs if needed
  - [ ] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [x ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by glasserc** [comment]
Looks OK. Add a test case for each, please?

**Comment by edi9999** [comment]
yes, +1 for tests

**Comment by epicfaace** [comment]
@althalus86 , do you still plan to work on this PR and add tests?

File: src/components/fields/ArrayField.js [hunk 1]
```diff
@@ -84,12 +87,12 @@ function DefaultArrayItem(props) {
             )}
 
             {props.hasRemove && (
-              <IconButton
-                type="danger"
-                icon="remove"
+              <RemoveButton
+                type={props.type || "danger"}
+                icon={props.icson || "remove"}
```
Comment (dachaac) [inline src/components/fields/ArrayField.js]:
```suggestion
                icon={props.icon || "remove"}
```

**Comment by dachaac** [comment]
Functionality here seems to be matching what we want to do (customisable & localisable button texts).

@althalus86 do you plan to conclude this PR or have this PR been abandoned?

**Comment by aldokkani** [comment]
Is this going to be merged any soon? I can help with any work left! @epicfaace

**Comment by heath-freenome** [comment]
Closing this in favor of https://github.com/rjsf-team/react-jsonschema-form/pull/3014

**Pull request closed without merging**